### PR TITLE
Add 3D-SSIM loss

### LINKED ISSUE #1158: [Feature request] Add 3D-SSIM
## 🚀 Feature
3D-SSIM

## Motivation
SSIM and SSIMLoss are currently only supported in 2D in kornia. While it is common in medical imaging to compute SSIM on 3D volumes. In addition, 3D-SSIM is useful for video.

## Pitch
Adding 3D-SSIM

## Alternatives
N/A

## Additional context
N/A

### PR DESCRIPTION
**Pull request by pri1311**
Hi team, this PR aims to add SSIM Loss and Metrics for 3D images. Based on the following comment - https://github.com/kornia/kornia/issues/1158#issuecomment-880569245 and the Monai Implementation. Will have to skim through the paper once more to follow up on this - https://github.com/kornia/kornia/issues/1158#issuecomment-880605633

Also, I just checked, the failing doctests are because of the recent changes in filters for gaussian kernels. Will make changes as per the new format.

Fixes #1158 

#### Type of change

- [ ] 📚  Documentation Update
- [ ] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

File: kornia/filters/kernels.py [hunk 1]
```diff
@@ -783,3 +783,49 @@ def get_hanning_kernel2d(kernel_size: Tuple[int, int], device=torch.device('cpu'
...
+    if not isinstance(kernel_size, tuple) or len(kernel_size) != 3:
+        raise TypeError(f"kernel_size must be a tuple of length three. Got {kernel_size}")
+    if not isinstance(sigma, tuple) or len(sigma) != 3:
+        raise TypeError(f"sigma must be a tuple of length three. Got {sigma}")
+    ksize_x, ksize_y, ksize_z = kernel_size
+    sigma_x, sigma_y, sigma_z = sigma
+    kernel_x: Tensor = get_gaussian_kernel1d(ksize_x, sigma_x, force_even)
+    kernel_y: Tensor = get_gaussian_kernel1d(ksize_y, sigma_y, force_even)
+    kernel_z: Tensor = get_gaussian_kernel1d(ksize_z, sigma_z, force_even)
+    kernel_2d: Tensor = torch.matmul(kernel_x.unsqueeze(-1), kernel_y.unsqueeze(-1).t())
```
Comment (edgarriba) [inline kornia/filters/kernels.py:829]:
```suggestion
    kernel_2d: Tensor = kernel_x[..., None] @ kernel_y[..., None].t()
```
more pythonic. same below

File: kornia/losses/ssim3d.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
+import torch
+import torch.nn as nn
+
+import kornia.metrics as metrics
+
+
+def ssim3d_loss(
+    img1: torch.Tensor,
```
Comment (edgarriba) [inline kornia/losses/ssim3d.py:8]:
```suggestion
    img1: Tensor,
```
`from kornia.core import Tensor`

File: kornia/losses/ssim3d.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    if reduction == "mean":
+        loss = torch.mean(loss)
+    elif reduction == "sum":
+        loss = torch.sum(loss)
+    elif reduction == "none":
+        pass
+    return loss
+
+
+class SSIM3DLoss(nn.Module):
```
Comment (edgarriba) [inline kornia/losses/ssim3d.py:62]:
```suggestion
class SSIM3DLoss(Module):
```
import fomr kornia core

File: kornia/metrics/ssim3d.py [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+
+    Returns:
+       The ssim index map with shape :math:`(B, C, D, H, W)`.
+
+    Examples:
+        >>> input1 = torch.rand(1, 4, 5, 5, 5)
+        >>> input2 = torch.rand(1, 4, 5, 5, 5)
+        >>> ssim_map = ssim(input1, input2, 5)  # 1x4x5x5x5
+    """
+    if not isinstance(img1, torch.Tensor):
```
Comment (edgarriba) [inline kornia/metrics/ssim3d.py:67]: use KORNIA_CHECK functions

**Review by edgarriba — COMMENTED** [review]
good direction. Missing tests

**Review by edgarriba — APPROVED** [review]

**Comment by edgarriba** [comment]
@pri1311 would be great if you can send a PR adding the added functions to docs
